### PR TITLE
5321: Get order summary

### PR DESCRIPTION
**Pull request by RuthBerry96**
[AB#6736](https://buyingcatalog.visualstudio.com/c5f97979-5b03-4d10-ba8d-871d0526b408/_workitems/edit/6736)

### CODE REVIEW

File: tests/NHSD.BuyingCatalogue.Ordering.Api.UnitTests/Controllers/OrdersControllerTests.cs [hunk 1]
```diff
@@ -117,6 +148,39 @@ private static (Order order, OrderModel expectedOrder) CreateOrderTestData(strin
...
+                .WithDescription(description)
+                .Build();
+
+            return (order: repositoryOrder,
+                expectedSummary: new OrderSummaryModel
+                {
+                    OrderId = repositoryOrder.OrderId,
+                    OrganisationId = repositoryOrder.OrganisationId,
+                    Description = repositoryOrder.Description,
+                    Sections = new List<SectionModel>()
```
Comment (bjss-chris) [inline tests/NHSD.BuyingCatalogue.Ordering.Api.UnitTests/Controllers/OrdersControllerTests.cs:165]: Brackets can be omitted here.

File: tests/NHSD.BuyingCatalogue.Ordering.Api.UnitTests/Controllers/OrdersControllerTests.cs [hunk 2]
```diff
@@ -127,19 +191,28 @@ private OrdersControllerTestContext()
...
             internal static OrdersControllerTestContext Setup()
             {
                 return new OrdersControllerTestContext();
             }
         }
+
+        private static void AssertSectionComplete(string status, bool shouldBeComplete)
+        {
+            status.Should().Be(shouldBeComplete ? "complete" : "incomplete");
+        }
```
Comment (bjss-chris) [inline tests/NHSD.BuyingCatalogue.Ordering.Api.UnitTests/Controllers/OrdersControllerTests.cs:216]: Do we need this?

File: src/NHSD.BuyingCatalogue.Ordering.Api/Controllers/OrdersController.cs [hunk 3]
```diff
@@ -47,25 +47,26 @@ public async Task<ActionResult> GetAllAsync(Guid organisationId)
```
Comment (bjss-chris) [inline src/NHSD.BuyingCatalogue.Ordering.Api/Controllers/OrdersController.cs:47]: I think there's an unnecessary using in this file – could you remove it while you're here?

File: tests/NHSD.BuyingCatalogue.Ordering.Api.IntegrationTests/Features/Order Summary 5321.feature [hunk 4]
```diff
@@ -0,0 +1,73 @@
...
+Scenario: 5. A non authority user cannot access the order summary
+	#This will have a non authority user logged in
+	When the user makes a request to retrieve the order summary with the ID C000014-01
+	Then a response with status code 403 is returned
+
+@5321
+Scenario: 6. Service Failure
+	Given the call to the database will fail
+	When the user makes a request to retrieve the order summary with the ID C000014-01
+	Then a response with status code 500 is returned
```
[the remark this answers is not in the document]
Comment (RuthBerry96) [reply tests/NHSD.BuyingCatalogue.Ordering.Api.IntegrationTests/Features/Order Summary 5321.feature:73]: 👍

File: tests/NHSD.BuyingCatalogue.Ordering.Api.IntegrationTests/Steps/OrderSummarySteps.cs [hunk 5]
```diff
@@ -0,0 +1,86 @@
...
+        }
+
+        [Then(@"the order summary is returned with the following values")]
+        public async Task ThenTheOrderSummaryIsReturnedWithTheFollowingValues(Table table)
+        {
+            var expected = table.CreateSet<OrderSummaryTable>().FirstOrDefault();
+
+            var response = await _response.ReadBodyAsJsonAsync();
+
+            var actual = new OrderSummaryTable()
```
[the remark this answers is not in the document]
Comment (RuthBerry96) [reply tests/NHSD.BuyingCatalogue.Ordering.Api.IntegrationTests/Steps/OrderSummarySteps.cs:43]: 👍

File: tests/NHSD.BuyingCatalogue.Ordering.Api.IntegrationTests/Steps/OrderSummarySteps.cs [hunk 6]
```diff
@@ -0,0 +1,86 @@
...
+        }
+
+        [Then(@"the order Summary Sections have the following values")]
+        public async Task ThenTheOrderSummarySectionsHaveTheFollowingValues(Table table)
+        {
+            var expected = table.CreateSet<SectionTable>();
+
+            var response = await _response.ReadBodyAsJsonAsync();
+
+            var actual = new SectionsTable()
```
[the remark this answers is not in the document]
Comment (RuthBerry96) [reply tests/NHSD.BuyingCatalogue.Ordering.Api.IntegrationTests/Steps/OrderSummarySteps.cs:60]: 👍

File: tests/NHSD.BuyingCatalogue.Ordering.Api.UnitTests/Controllers/OrdersControllerTests.cs [hunk 7]
```diff
@@ -96,6 +97,36 @@ public async Task GetAll_OrdersByOrganisationId_CalledOnce()
...
+            var response = await controller.GetOrderSummaryAsync("INVALID");
+            response.Should().BeEquivalentTo(new NotFoundResult());
+        }
+
+        [TestCase("")]
+        [TestCase(" ")]
+        [TestCase("Some Description")]
+        public async Task GetOrderSummaryAsync_IsSummaryComplete_ReturnResult(string description)
+        {
+            var orderId = "C0000014-01";
```
[the remark this answers is not in the document]
Comment (RuthBerry96) [reply tests/NHSD.BuyingCatalogue.Ordering.Api.UnitTests/Controllers/OrdersControllerTests.cs:116]: 👍

File: tests/NHSD.BuyingCatalogue.Ordering.Api.UnitTests/Controllers/OrdersControllerTests.cs (hunk 1 above)
Comment (RuthBerry96) [reply tests/NHSD.BuyingCatalogue.Ordering.Api.UnitTests/Controllers/OrdersControllerTests.cs:165]: 👍

File: tests/NHSD.BuyingCatalogue.Ordering.Api.UnitTests/Controllers/OrdersControllerTests.cs (hunk 2 above)
Comment (RuthBerry96) [reply tests/NHSD.BuyingCatalogue.Ordering.Api.UnitTests/Controllers/OrdersControllerTests.cs:216]: Ahh no! I was going to use something like this and must have forgot to remove it when I decided against it. Thanks

File: src/NHSD.BuyingCatalogue.Ordering.Api/Controllers/OrdersController.cs (hunk 3 above)
Comment (RuthBerry96) [reply src/NHSD.BuyingCatalogue.Ordering.Api/Controllers/OrdersController.cs:47]: 👍

File: tests/NHSD.BuyingCatalogue.Ordering.Api.UnitTests/Controllers/OrdersControllerTests.cs [hunk 8]
```diff
@@ -96,6 +97,36 @@ public async Task GetAll_OrdersByOrganisationId_CalledOnce()
...
+
+            var order = CreateOrderSummaryTestData(orderId, description);
```
[the remark this answers is not in the document]
Comment (RuthBerry96) [reply tests/NHSD.BuyingCatalogue.Ordering.Api.UnitTests/Controllers/OrdersControllerTests.cs:118]: Sounds good 😄

File: tests/NHSD.BuyingCatalogue.Ordering.Api.IntegrationTests/Steps/OrderSummarySteps.cs [hunk 9]
```diff
@@ -0,0 +1,86 @@
...
+using TechTalk.SpecFlow.Assist;
+
+namespace NHSD.BuyingCatalogue.Ordering.Api.IntegrationTests.Steps
+{
+    [Binding]
+    internal sealed class OrderSummarySteps
+    {
+        private readonly Response _response;
+        private readonly Request _request;
+        private readonly Settings _settings;
```
[the remark this answers is not in the document]
Comment (RuthBerry96) [reply tests/NHSD.BuyingCatalogue.Ordering.Api.IntegrationTests/Steps/OrderSummarySteps.cs:18]: 👍

**Pull request merged**